src/api_server/src/http.rs

```diff
@@ -162,11 +163,37 @@ pub(crate) async fn resource(
                         format!("{resource_repo}/{resource_type}/{resource_tag}");
                     resource_path.push(ref_resource_path);
 
-                    if let Ok(resource_byte) = ::std::fs::read(&resource_path) {
-                        // TODO
-                        return HttpResponse::NotFound().finish();
-                    } else {
-                        return HttpResponse::NotFound().finish();
+                    match ::std::fs::read(&resource_path) {
+                        Ok(resource_byte) => match session.tee_public_key() {
```

Comment:
Adding a couple of `Session` methods: `secret_resource()` and `token_resource()` that would call into the `resource.rs` API and return an HTTP response would make the code below more compact and easier to read.